File: backend/src/main/java/t/lab/guide/dto/admin/point/AdminPointMediaItem.java
package t.lab.guide.dto.admin.point;

import io.swagger.v3.oas.annotations.media.Schema;
import lombok.Builder;
import t.lab.guide.entity.enums.MediaType;

import java.time.OffsetDateTime;

@Schema(description = "Медиа-материал точки интереса в административном представлении (с идентификатором для управления)")
@Builder
public record AdminPointMediaItem(
        @Schema(description = "Уникальный идентификатор медиа-материала", example = "10")
        Long id,
        
        @Schema(description = "Идентификатор точки, к которому отновится media")
        Long pointId,

        @Schema(description = "Ссылка на медиа-файл",
                example = "https://cdn.t-guide.mock/points/1/photo-1.jpg")
        String url,

        @Schema(description = "Тип медиа-материала", example = "PHOTO")
        MediaType type,

        @Schema(description = "Порядковый номер для сортировки", example = "0")
        Integer sortOrder,

        @Schema(description = "Дата и время загрузки медиа", example = "2024-01-01T12:00:00Z")
        OffsetDateTime createdAt
) {}

File: backend/src/main/kotlin/t/lab/guide/dto/admin/point/AdminPointMediaItem.kt
package t.lab.guide.dto.admin.point

import io.swagger.v3.oas.annotations.media.Schema
import t.lab.guide.enums.MediaType
import java.time.OffsetDateTime

@Schema(description = "Медиа-материал точки интереса в административном представлении (с идентификатором для управления)")
data class AdminPointMediaItem(
    @Schema(description = "Уникальный идентификатор медиа-материала", example = "10")
    val id: Long,
    @Schema(description = "Идентификатор точки, к которому относится media")
    val pointId: Long,
    @Schema(
        description = "Ссылка на медиа-файл",
        example = "https://cdn.t-guide.mock/points/1/photo-1.jpg",
    )
    val url: String,
    @Schema(description = "Тип медиа-материала", example = "PHOTO")
    val type: MediaType,
    @Schema(description = "Порядковый номер для сортировки", example = "0")
    val sortOrder: Int,
    @Schema(description = "Дата и время загрузки медиа", example = "2024-01-01T12:00:00Z")
    val createdAt: OffsetDateTime,
)
